### ISSUE TITLE
Can't even load the mod without crashing

### ISSUE THREAD

**Issue by ImadeThisJustToReportThis** [Wilyicaro/Legacy-Minecraft#487]
everytime I try and open the game with this mod, it crashes and I dont know why?

it keeps showing exit code 1

**Comment by Pasithea0** [comment]
I had that same issue, but disabling ImmediatelyFast mod fixed it.

**Comment by LapisPhoenix** [comment]
I have the same issue, Linux Mint. I'm using Neoforge for Minecraft version 1.20.4, but was still happening with 1.20.6. I also was previously using Fabric for Minecraft 1.20.6, same behavior. Here is my part latest log:

```
[30Jun2024 12:45:39.319] [Render thread/INFO] [net.minecraft.server.packs.resources.ReloadableResourceManager/]: Reloading ResourceManager: mod_resources, mod:legacy, mod:architectury, mod:neoforge, vanilla
[30Jun2024 12:45:39.379] [modloading-worker-0/WARN] [net.neoforged.neoforge.common.ModConfigSpec/CORE]: Configuration file /home/bleh/.minecraft/config/neoforge-client.toml is not correct. Correcting
[30Jun2024 12:45:39.380] [modloading-worker-0/WARN] [net.neoforged.neoforge.common.ModConfigSpec/CORE]: Incorrect key client was corrected from null to its default, SimpleCommentedConfig:{}. 
[30Jun2024 12:45:39.381] [modloading-worker-0/WARN] [net.neoforged.neoforge.common.ModConfigSpec/CORE]: Incorrect key client.experimentalForgeLightPipelineEnabled was corrected from null to its default, false. 
[30Jun2024 12:45:39.381] [modloading-worker-0/WARN] [net.neoforged.neoforge.common.ModConfigSpec/CORE]: Incorrect key client.showLoadWarnings was corrected from null to its default, true. 
[30Jun2024 12:45:39.381] [modloading-worker-0/WARN] [net.neoforged.neoforge.common.ModConfigSpec/CORE]: Incorrect key client.useCombinedDepthStencilAttachment was corrected from null to its default, false. 
[30Jun2024 12:45:39.381] [modloading-worker-0/WARN] [net.neoforged.neoforge.common.ModConfigSpec/CORE]: Incorrect key client.compressLanIPv6Addresses was corrected from null to its default, true. 
[30Jun2024 12:45:39.405] [Worker-Main-3/INFO] [net.minecraft.client.gui.font.providers.UnihexProvider/]: Found unifont_all_no_pua-15.1.04.hex, loading
[30Jun2024 12:45:40.015] [Render thread/ERROR] [net.minecraft.client.main.Main/]: Unhandled game exception
java.lang.UnsatisfiedLinkError: Error looking up function 'SDL_GameControllerGetFirmwareVersion': /lib/x86_64-linux-gnu/libSDL2.so: undefined symbol: SDL_GameControllerGetFirmwareVersion
	at com.sun.jna.Function.<init>(Function.java:252) ~[jna-5.13.0.jar%23146!/:5.13.0 (b0)]
	at com.sun.jna.NativeLibrary.getFunction(NativeLibrary.java:620) ~[jna-5.13.0.jar%23146!/:5.13.0 (b0)]
	at com.sun.jna.NativeLibrary.getFunction(NativeLibrary.java:596) ~[jna-5.13.0.jar%23146!/:5.13.0 (b0)]
	at com.sun.jna.NativeLibrary.getFunction(NativeLibrary.java:582) ~[jna-5.13.0.jar%23146!/:5.13.0 (b0)]
	at com.sun.jna.Native.register(Native.java:1904) ~[jna-5.13.0.jar%23146!/:5.13.0 (b0)]
	at io.github.libsdl4j.jna.SdlNativeLibraryLoader.registerNativeMethods(SdlNativeLibraryLoader.java:26) ~[libsdl4j-2.28.4-1.6.jar%23175!/:?]
	at io.github.libsdl4j.api.gamecontroller.SdlGamecontroller.<clinit>(SdlGamecontroller.java:69) ~[libsdl4j-2.28.4-1.6.jar%23175!/:?]
	at wily.legacy.client.controller.SDLControllerHandler.applyGamePadMappingsFromBuffer(SDLControllerHandler.java:136) ~[Legacy4J-1.20.4-1.6.7-neoforge.jar%23170!/:?]
	at wily.legacy.client.controller.Controller$Handler.tryDownloadAndApplyNewMappings(Controller.java:62) ~[Legacy4J-1.20.4-1.6.7-neoforge.jar%23170!/:?]
	at wily.legacy.client.controller.SDLControllerHandler.init(SDLControllerHandler.java:33) ~[Legacy4J-1.20.4-1.6.7-neoforge.jar%23170!/:?]
	at wily.legacy.client.controller.ControllerManager.setup(ControllerManager.java:55) ~[Legacy4J-1.20.4-1.6.7-neoforge.jar%23170!/:?]
	at wily.legacy.Legacy4JClient.lambda$init$14(Legacy4JClient.java:253) ~[Legacy4J-1.20.4-1.6.7-neoforge.jar%23170!/:?]
	at net.minecraft.client.Minecraft.handler$zhp000$architectury$onStart(Minecraft.java:4146) ~[client-1.20.4-20231207.154220-srg.jar%23171!/:?]
	at net.minecraft.client.Minecraft.run(Minecraft.java:802) ~[client-1.20.4-20231207.154220-srg.jar%23171!/:?]
	at net.minecraft.client.main.Main.main(Main.java:244) ~[neoforge-20.4.237.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at net.neoforged.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:136) ~[loader-2.0.17.jar:2.0]
	at net.neoforged.fml.loading.targets.CommonLaunchHandler.clientService(CommonLaunchHandler.java:124) ~[loader-2.0.17.jar:2.0]
	at net.neoforged.fml.loading.targets.CommonClientLaunchHandler.runService(CommonClientLaunchHandler.java:24) ~[loader-2.0.17.jar:2.0]
	at net.neoforged.fml.loading.targets.CommonLaunchHandler.lambda$launchService$4(CommonLaunchHandler.java:118) ~[loader-2.0.17.jar:2.0]
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:108) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:78) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141) ~[bootstraplauncher-1.1.2.jar:?]
[30Jun2024 12:45:40.018] [Render thread/INFO] [net.minecraft.client.Minecraft/]: Stopping!
[30Jun2024 12:45:40.177] [Render thread/ERROR] [net.minecraft.client.Minecraft/]: Shutdown failure!
java.util.ConcurrentModificationException: null
	at java.util.HashMap.forEach(HashMap.java:1424) ~[?:?]
	at net.minecraft.client.renderer.texture.TextureManager.close(TextureManager.java:173) ~[client-1.20.4-20231207.154220-srg.jar%23171!/:?]
	at net.minecraft.client.Minecraft.close(Minecraft.java:1190) ~[client-1.20.4-20231207.154220-srg.jar%23171!/:?]
	at net.minecraft.client.Minecraft.destroy(Minecraft.java:1163) ~[client-1.20.4-20231207.154220-srg.jar%23171!/:?]
	at net.minecraft.client.main.Main.main(Main.java:260) ~[neoforge-20.4.237.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at net.neoforged.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:136) ~[loader-2.0.17.jar:2.0]
	at net.neoforged.fml.loading.targets.CommonLaunchHandler.clientService(CommonLaunchHandler.java:124) ~[loader-2.0.17.jar:2.0]
	at net.neoforged.fml.loading.targets.CommonClientLaunchHandler.runService(CommonClientLaunchHandler.java:24) ~[loader-2.0.17.jar:2.0]
	at net.neoforged.fml.loading.targets.CommonLaunchHandler.lambda$launchService$4(CommonLaunchHandler.java:118) ~[loader-2.0.17.jar:2.0]
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:108) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:78) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141) ~[bootstraplauncher-1.1.2.jar:?]
```

https://github.com/Wilyicaro/Legacy-Minecraft/assets/88260244/e8038cab-fcdb-40d8-90e4-e44d35e6b22a

**Comment by LapisPhoenix** [comment]
> I have the same issue, Linux Mint. I'm using Neoforge for Minecraft version 1.20.4, but was still happening with 1.20.6. I also was previously using Fabric for Minecraft 1.20.6, same behavior. Here is part of my latest .log file:
> 
> Seems to be a problem with `SDL_GameControllerGetFirmwareVersion` method?
> 
> ```
> [30Jun2024 12:45:39.319] [Render thread/INFO] [net.minecraft.server.packs.resources.ReloadableResourceManager/]: Reloading ResourceManager: mod_resources, mod:legacy, mod:architectury, mod:neoforge, vanilla
> [30Jun2024 12:45:39.379] [modloading-worker-0/WARN] [net.neoforged.neoforge.common.ModConfigSpec/CORE]: Configuration file /home/bleh/.minecraft/config/neoforge-client.toml is not correct. Correcting
> [30Jun2024 12:45:39.380] [modloading-worker-0/WARN] [net.neoforged.neoforge.common.ModConfigSpec/CORE]: Incorrect key client was corrected from null to its default, SimpleCommentedConfig:{}. 
> [30Jun2024 12:45:39.381] [modloading-worker-0/WARN] [net.neoforged.neoforge.common.ModConfigSpec/CORE]: Incorrect key client.experimentalForgeLightPipelineEnabled was corrected from null to its default, false. 
> [30Jun2024 12:45:39.381] [modloading-worker-0/WARN] [net.neoforged.neoforge.common.ModConfigSpec/CORE]: Incorrect key client.showLoadWarnings was corrected from null to its default, true. 
> [30Jun2024 12:45:39.381] [modloading-worker-0/WARN] [net.neoforged.neoforge.common.ModConfigSpec/CORE]: Incorrect key client.useCombinedDepthStencilAttachment was corrected from null to its default, false. 
> [30Jun2024 12:45:39.381] [modloading-worker-0/WARN] [net.neoforged.neoforge.common.ModConfigSpec/CORE]: Incorrect key client.compressLanIPv6Addresses was corrected from null to its default, true. 
> [30Jun2024 12:45:39.405] [Worker-Main-3/INFO] [net.minecraft.client.gui.font.providers.UnihexProvider/]: Found unifont_all_no_pua-15.1.04.hex, loading
> [30Jun2024 12:45:40.015] [Render thread/ERROR] [net.minecraft.client.main.Main/]: Unhandled game exception
> java.lang.UnsatisfiedLinkError: Error looking up function 'SDL_GameControllerGetFirmwareVersion': /lib/x86_64-linux-gnu/libSDL2.so: undefined symbol: SDL_GameControllerGetFirmwareVersion
> 	at com.sun.jna.Function.<init>(Function.java:252) ~[jna-5.13.0.jar%23146!/:5.13.0 (b0)]
> 	at com.sun.jna.NativeLibrary.getFunction(NativeLibrary.java:620) ~[jna-5.13.0.jar%23146!/:5.13.0 (b0)]
> 	at com.sun.jna.NativeLibrary.getFunction(NativeLibrary.java:596) ~[jna-5.13.0.jar%23146!/:5.13.0 (b0)]
> 	at com.sun.jna.NativeLibrary.getFunction(NativeLibrary.java:582) ~[jna-5.13.0.jar%23146!/:5.13.0 (b0)]
> 	at com.sun.jna.Native.register(Native.java:1904) ~[jna-5.13.0.jar%23146!/:5.13.0 (b0)]
> 	at io.github.libsdl4j.jna.SdlNativeLibraryLoader.registerNativeMethods(SdlNativeLibraryLoader.java:26) ~[libsdl4j-2.28.4-1.6.jar%23175!/:?]
> 	at io.github.libsdl4j.api.gamecontroller.SdlGamecontroller.<clinit>(SdlGamecontroller.java:69) ~[libsdl4j-2.28.4-1.6.jar%23175!/:?]
> 	at wily.legacy.client.controller.SDLControllerHandler.applyGamePadMappingsFromBuffer(SDLControllerHandler.java:136) ~[Legacy4J-1.20.4-1.6.7-neoforge.jar%23170!/:?]
> 	at wily.legacy.client.controller.Controller$Handler.tryDownloadAndApplyNewMappings(Controller.java:62) ~[Legacy4J-1.20.4-1.6.7-neoforge.jar%23170!/:?]
> 	at wily.legacy.client.controller.SDLControllerHandler.init(SDLControllerHandler.java:33) ~[Legacy4J-1.20.4-1.6.7-neoforge.jar%23170!/:?]
> 	at wily.legacy.client.controller.ControllerManager.setup(ControllerManager.java:55) ~[Legacy4J-1.20.4-1.6.7-neoforge.jar%23170!/:?]
> 	at wily.legacy.Legacy4JClient.lambda$init$14(Legacy4JClient.java:253) ~[Legacy4J-1.20.4-1.6.7-neoforge.jar%23170!/:?]
> 	at net.minecraft.client.Minecraft.handler$zhp000$architectury$onStart(Minecraft.java:4146) ~[client-1.20.4-20231207.154220-srg.jar%23171!/:?]
> 	at net.minecraft.client.Minecraft.run(Minecraft.java:802) ~[client-1.20.4-20231207.154220-srg.jar%23171!/:?]
> 	at net.minecraft.client.main.Main.main(Main.java:244) ~[neoforge-20.4.237.jar:?]
> 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
> 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
> 	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
> 	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
> 	at net.neoforged.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:136) ~[loader-2.0.17.jar:2.0]
> 	at net.neoforged.fml.loading.targets.CommonLaunchHandler.clientService(CommonLaunchHandler.java:124) ~[loader-2.0.17.jar:2.0]
> 	at net.neoforged.fml.loading.targets.CommonClientLaunchHandler.runService(CommonClientLaunchHandler.java:24) ~[loader-2.0.17.jar:2.0]
> 	at net.neoforged.fml.loading.targets.CommonLaunchHandler.lambda$launchService$4(CommonLaunchHandler.java:118) ~[loader-2.0.17.jar:2.0]
> 	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) ~[modlauncher-10.0.9.jar:?]
> 	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) ~[modlauncher-10.0.9.jar:?]
> 	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) ~[modlauncher-10.0.9.jar:?]
> 	at cpw.mods.modlauncher.Launcher.run(Launcher.java:108) ~[modlauncher-10.0.9.jar:?]
> 	at cpw.mods.modlauncher.Launcher.main(Launcher.java:78) ~[modlauncher-10.0.9.jar:?]
> 	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) ~[modlauncher-10.0.9.jar:?]
> 	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) ~[modlauncher-10.0.9.jar:?]
> 	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141) ~[bootstraplauncher-1.1.2.jar:?]
> [30Jun2024 12:45:40.018] [Render thread/INFO] [net.minecraft.client.Minecraft/]: Stopping!
> [30Jun2024 12:45:40.177] [Render thread/ERROR] [net.minecraft.client.Minecraft/]: Shutdown failure!
> java.util.ConcurrentModificationException: null
> 	at java.util.HashMap.forEach(HashMap.java:1424) ~[?:?]
> 	at net.minecraft.client.renderer.texture.TextureManager.close(TextureManager.java:173) ~[client-1.20.4-20231207.154220-srg.jar%23171!/:?]
> 	at net.minecraft.client.Minecraft.close(Minecraft.java:1190) ~[client-1.20.4-20231207.154220-srg.jar%23171!/:?]
> 	at net.minecraft.client.Minecraft.destroy(Minecraft.java:1163) ~[client-1.20.4-20231207.154220-srg.jar%23171!/:?]
> 	at net.minecraft.client.main.Main.main(Main.java:260) ~[neoforge-20.4.237.jar:?]
> 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
> 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
> 	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
> 	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
> 	at net.neoforged.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:136) ~[loader-2.0.17.jar:2.0]
> 	at net.neoforged.fml.loading.targets.CommonLaunchHandler.clientService(CommonLaunchHandler.java:124) ~[loader-2.0.17.jar:2.0]
> 	at net.neoforged.fml.loading.targets.CommonClientLaunchHandler.runService(CommonClientLaunchHandler.java:24) ~[loader-2.0.17.jar:2.0]
> 	at net.neoforged.fml.loading.targets.CommonLaunchHandler.lambda$launchService$4(CommonLaunchHandler.java:118) ~[loader-2.0.17.jar:2.0]
> 	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) ~[modlauncher-10.0.9.jar:?]
> 	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) ~[modlauncher-10.0.9.jar:?]
> 	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) ~[modlauncher-10.0.9.jar:?]
> 	at cpw.mods.modlauncher.Launcher.run(Launcher.java:108) ~[modlauncher-10.0.9.jar:?]
> 	at cpw.mods.modlauncher.Launcher.main(Launcher.java:78) ~[modlauncher-10.0.9.jar:?]
> 	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) ~[modlauncher-10.0.9.jar:?]
> 	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) ~[modlauncher-10.0.9.jar:?]
> 	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141) ~[bootstraplauncher-1.1.2.jar:?]
> ```
> 
> 2024-06-30.12-54-57.mp4

Well I was able to fix it, I ran `sudo apt-get remove libsdl2-dev` and now it launches without issue, found solution from [here](https://github.com/MrCrayfish/Controllable/issues/408#issuecomment-1606361273)

**Comment by Wilyicaro** [comment]
Fixed in version 1.7, thanks for reporting.
